### PR TITLE
Small subtree cache optimization.

### PR DESCRIPTION
**Pull request by Martin2112**
Don't calculate some things unless we need to use them. Probably not a huge win but we might as well have it as it's slightly clearer to read.

### Checklist

- [ ] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [ ] I have updated [documentation](docs/) accordingly.

### CODE REVIEW

File: storage/cache/log_subtree_cache.go [hunk 1]
```diff
@@ -74,13 +74,13 @@ func populateLogSubtreeNodes(hasher hashers.LogHasher) storage.PopulateSubtreeFu
...
 
-				subDepth := logStrataDepth - height
-				nodeID := storage.NewNodeIDFromPrefix(st.Prefix, subDepth, index, logStrataDepth, maxLogDepth)
-				_, sfx := nodeID.Split(len(st.Prefix), int(st.Depth))
-				sfxKey := sfx.String()
 				// Don't put leaves into the internal map and only update if we're rebuilding internal
 				// nodes. If the subtree was saved with internal nodes then we don't touch the map.
 				if height > 0 && len(st.Leaves) == maxLeaves {
+					subDepth := logStrataDepth - height
+					nodeID := storage.NewNodeIDFromPrefix(st.Prefix, subDepth, index, logStrataDepth, maxLogDepth)
```
Comment (AlCutter) [inline storage/cache/log_subtree_cache.go]: _Might_ be worth seeing if there's a cheaper approach to figuring out the internal node suffix?

`NewNodeIDFromPrefix` looks quite heavyweight, and we don't care about the prefix here.

Comment (Martin2112) [reply storage/cache/log_subtree_cache.go]: Yep, was trying to limit the frightfulness factor for the moment. TODO it?

Comment (Martin2112) [reply storage/cache/log_subtree_cache.go]: I think we should micro benchmark this code before more serious optimizations.

Comment (AlCutter) [reply storage/cache/log_subtree_cache.go]: Sure :)

Comment (Martin2112) [reply storage/cache/log_subtree_cache.go]: I suspect the cost of hashing dominates but it would be good to know.

Comment (Martin2112) [reply storage/cache/log_subtree_cache.go]: TODO added.

**Pull request merged**